src/main/java/com/cburch/logisim/fpga/gui/FpgaIoInformationSettingsDialog.java

```diff
@@ -712,11 +714,17 @@ else if (cnt < nrOfPins.get(INPUT_ID) + nrOfPins.get(OUTPUT_ID))
     CancelButton.addActionListener(actionListener);
     gbc.gridx = 0;
     contents.add(CancelButton, gbc);
+    // FIXME: Find better solution to handle window close button
+    selWindow.addWindowListener(new WindowAdapter() {
+      @Override
+      public void windowClosing(WindowEvent e) {
+        CancelButton.doClick();
+      }
+    });
     selWindow.add(new JScrollPane(contents));
     selWindow.pack();
     selWindow.setLocationRelativeTo(IOcomps.getParentFrame());
     selWindow.setModal(true);
-    selWindow.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
```

Comment:
Why you delete it here and you don't do the same in line 1102 below (you need to expand the view)?

Reply:
Looking into it I probably should not have removed it. Both of the 'x' buttons work, so having it in there is not breaking the new functionality. I will add it back.
I have not worked with GUIs before.